iDiff/utils/tar_utils_test.go

```diff
@@ -36,21 +40,112 @@ func TestUnTar(t *testing.T) {
 			target:   "testTars/la-croix3",
 			expected: "testTars/la-croix3-actual",
 		},
+		{
+			descrip:  "Tar with .wh.'s",
+			tarPath:  "testTars/la-croix-wh.tar",
+			target:   "testTars/la-croix-wh",
+			expected: "testTars/la-croix-wh-actual",
+			starter:  "testTars/la-croix-starter",
+		},
+		{
+			descrip:  "Files updated",
+			tarPath:  "testTars/la-croix-update.tar",
+			target:   "testTars/la-croix-update",
+			expected: "testTars/la-croix-update-actual",
+			starter:  "testTars/la-croix-starter",
+		},
 	}
 	for _, test := range testCases {
+		if test.starter != "" {
+			CopyDir(test.starter, test.target)
+		}
 		err := UnTar(test.tarPath, test.target)
 		if err != nil && !test.err {
-			t.Errorf("Got unexpected error: %s", err)
+			t.Errorf(test.descrip, "Got unexpected error: %s", err)
 		}
 		if err == nil && test.err {
-			t.Errorf("Expected error but got none: %s", err)
+			t.Errorf(test.descrip, "Expected error but got none: %s", err)
 		}
 		if !dirEquals(test.expected, test.target) || !dirEquals(test.target, test.expected) {
-			t.Errorf("Directory created not correct structure.")
+			t.Errorf(test.descrip, "Directory created not correct structure.")
 		}
 		os.RemoveAll(test.target)
 	}
+}
+
+// Copies file source to destination dest.
+func CopyFile(source string, dest string) (err error) {
+	sf, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+	defer sf.Close()
+	df, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer df.Close()
+	_, err = io.Copy(df, sf)
+	if err == nil {
+		si, err := os.Stat(source)
+		if err != nil {
+			err = os.Chmod(dest, si.Mode())
+		}
+
+	}
+
+	return nil
+}
 
+// Recursively copies a directory tree, attempting to preserve permissions.
+// Source directory must exist, destination directory must *not* exist.
+func CopyDir(source string, dest string) (err error) {
+
+	// get properties of source dir
+	fi, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+
+	if !fi.IsDir() {
+		return fmt.Errorf("Source not a directory")
```

Comment:
In other places in iDiff, the errors pkg is used (errors.New("")) in place of fmt.Errorf.  The benefit of using errors is error chaining and stacktraces.  The error package used should be consistent.